website/docs/developer/research/research-lambda-container.md

```diff
@@ -0,0 +1,139 @@
+- Image configuration- These are values that can be used to override the container image settings for `ENTRYPOINT`, `CMD`, `WORKDIR` and `ENV`.
+
+```
+  ENTRYPOINT – Specifies the absolute path of the entry point to the application.
+
+  CMD – Specifies parameters that you want to pass in with ENTRYPOINT.
+
+  WORKDIR – Specifies the absolute path of the working directory.
+
+  ENV – Specifies an environment variable for the Lambda function.
+```
+- Image URI- The location of the container image to use for your function.
+
+
+## Creating a prototype
+Creating or updating the function is done by building a Docker image, uploading the new version to ECR and deploying/updating the Lambda function to point to the newly uploaded image using terraform. Docker CLI has been used to perform the task.
+
+The steps for prototyping lambda container are as follows:
+- Create or use an existing AWS ECR repository to store the Docker image. 
+- Create the Dockerfile and then build, tag and push the image to the above ECR repository.
+- Update terraform configuration of the lambda function to deploy it as container.
+
+:::warning
+Currently NGAP only allows developers to create a `private` ECR repository which was used here to create the prototype. Using a `public` ECR repository will need approval from NGAP first which could bring security concerns.
+:::
+
+Details on creating the prototype are shown below:
+
+1. Create an ECR repository from AWS CLI if needed as shown. Check [here](https://docs.aws.amazon.com/AmazonECR/latest/userguide/getting-started-cli.html#cli-create-repository) for additional details on this.
+
+```bash
+aws ecr create-repository \
+    --repository-name <YOUR_REPOSITORY_NAME>
+```
+
+2. Create a project directory <MY_DIR>. Under the directory, add  your script (`test.py` in this case) and `requirements.txt`to install any dependencies. Then  create a `Dockerfile` that creates the image. An example of a Dockerfile used for prototying a lambda having `test.py` file is shown.
+
+```yaml
+FROM public.ecr.aws/lambda/python:3.8
+
+# Copy function code
+COPY test.py ${LAMBDA_TASK_ROOT}
+
+# Install the function's dependencies using file requirements.txt
+# from your project folder.
+
+COPY requirements.txt  .
+RUN  pip3 install -r requirements.txt --target "${LAMBDA_TASK_ROOT}"
+
+# Set the CMD to your handler (could also be done as a parameter override outside of the Dockerfile)
+CMD [ "test.hambda_handler" ]
+
+```
+
+3. The next steps are to build, tag and push the image to the ECR repo. You can build the image which is named  `prototype-lambda-image` in this case using:
+
+```bash
+docker build -t prototype-lambda-image .
+```
+Once build is successful, tag the image
+
+```bash
+docker tag prototype-lambda-image:latest <YOUR_ECR_REPO_URI>:latest
+```
+Next, login to the ECR repo using:
+
+```bash
+aws ecr get-login-password --region <YOUR_REGION> | docker login --username AWS --password<YOUR_PASSWORD>
```

Comment:
Probably need a space before "<YOUR_PASSWORD>"

Reply:
might look strange but this is how it's written in the cli and works.

Reply:
Closest I can find is [this](https://docs.aws.amazon.com/cli/latest/reference/ecr/get-login-password.html) which doesn't match well. Mind sharing your source? What is your typed command?

Reply:
The examples I've found either use `--password-stdin` or `--password=somevalue`. Are you saying the script is literally written as `--password<YOUR_PASSWORD>`? If so, we should revisit how other variable values are defined, such as <YOUR_ECR_REPO_URI>.

Reply:
looks like it's my fault.

it should actually be `--password-stdin` after taking a second look.

updated.

Reply:
I believe you can remove the <YOUR_PASSWORD> since it is provided by standard input. That said, haven't run this myself.

Reply:
got this error without <YOUR_PASSWORD>

```

IGSKMNCNLTM0057:cumulus-orca rhassan$ aws ecr get-login-password --region us-west-2 | docker login --username AWS --password-stdin
Error response from daemon: Get "https://registry-1.docker.io/v2/": unauthorized: incorrect username or password
```

Reply:
Is <YOUR_PASSWORD> a magic string, or does the user replace that?
If the former, I haven't found docs around it.
If the later, the point of --password-stdin is to not have to type a password into a console in such a way that it will appear in logs.

Reply:
the user has to replace that

check this example 
https://docs.aws.amazon.com/cli/latest/reference/ecr/get-login-password.html#examples

Reply:
I see no manually entered password in this example.

Reply:
<aws_account_id>.dkr.ecr.<region>.amazonaws.com

Reply:
Ah. That is not a password. That is the ECR Repository URI. I t just happens to come after the --password-stdin argument.

Reply:
aah I see what's hapening. We don't actaully provide any password.

I will update the sentence and remove the password stuff